lib/id3c/cli/command/etl/manifest.py

```diff
@@ -153,9 +170,12 @@ def etl_manifest(*, db: DatabaseSession):
 
 
 def upsert_sample(db: DatabaseSession,
+                  record_has_sample: bool,
+                  record_has_collection: bool,
+                  record_has_collected_date: bool,
```

Comment:
There is an important distinction I'm making here. I want to make sure this is getting across.

These parameters are passing in the values pulled out of the record. They may be real values or None.
identifier: Optional[str],
collection_identifier: Optional[str],
collected_date: Optional[str],
additional_details: dict) -> Tuple[Any, str]:

These 3 new parameters are saying whether the key existed in the record at all.
record_has_sample: bool,
record_has_collection: bool,
record_has_collected_date: bool,

From looking only at collection_identifier, for example, you can't tell whether the manifest settings tried to extract collection_identifier from the record and that record happened not to have a value for it or if the manifest wasn't configured to try to pull out collection_identifier. 

The SQL logic is driven by whether the keys existed in the record, not by whether the columns had a non-None value for the keys.

Does that explain why the new parameters are needed?
If there is a  tri-state data type that would combine these into one sec of parameters let me know. I could imagine the three states as Not Set (the manifest didn't even try), None, and a real value.

Reply:
I think the SQL logic _should_ be driven by a non-None value. I don't think we want to update any existing value to `None` especially once we also ingest collected date from REDCap.

Reply:
I was on the same train of thought with you with the previous implementation when I had the coalesce's in there.

If I made the update dependent on whether the record had a non-None value for the field this would be a change from current behavior. If an aliquot record came in with collection ABC and sample 123 and then later they updated the sheet removing the sample ID, after this change we would keep the collection mapped to the sample.  

Is that what we want?

Reply:
I agree with @joverlee521 re: creating the boolean flags within `upsert_sample` rather than passing them in as additional parameters. In my opinion, it's better practice to include fewer parameters and equip functions with the knowledge to parse a document we pass it. We try to employ this in other areas of id3c/id3c-customizations as well.

Reply:
I've gone with coalescing the identifier and collection_identifier columns. Thomas recently added in the coalesce in the update for collected. This obviates those flag parameters and the SQL command string building. FYI @joverlee521